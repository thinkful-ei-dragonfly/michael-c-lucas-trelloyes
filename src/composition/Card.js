import React from 'react';
import './Card.css';

function Card(props) {
  return (
    <div className='Card'>
      <button type='button'>delete</button>
      <h3>{props.title}</h3>
      <p>{props.content}</p>
    </div>
  )
}

class CardClass extends React.Component {
  static defaultProps = {
    title: '',
    content: ''
  };
  render(){
    return (
      <div className='Card'>
        <button type='button'>delete</button>
        <h3>{this.props.title}</h3>
        <p>{this.props.content}</p>
      </div>
    )
  }
}

export default Card;
