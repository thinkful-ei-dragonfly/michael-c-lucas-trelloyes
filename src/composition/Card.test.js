import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Card from './Card';

describe('<Card />', () => {
  it('renders without crashing [snapshot test]', () => {
    // Render this component as JSON
    const tree = renderer.create(<Card/>).toJSON();
    // Checks wether this matches the previous snapshot
    // The previous snapshot will be stored in __snapshots__/Card.test.js
    expect(tree).toMatchSnapshot();
  })
})
