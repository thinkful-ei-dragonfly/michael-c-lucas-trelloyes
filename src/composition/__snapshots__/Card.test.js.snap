// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders without crashing [snapshot test] 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    
  </h3>
  <p>
    
  </p>
</div>
`;
